x/posts/types/models/common/post_optional_data.go

```diff
@@ -1,65 +1,30 @@
 package common
 
-import (
-	"encoding/json"
-	"sort"
-)
+import "fmt"
 
 // OptionalData represents a Posts' optional data and allows for custom
 // Amino and JSON serialization and deserialization.
-type OptionalData map[string]string
-
-// KeyValue is a simple key/value representation of one field of a OptionalData.
-type KeyValue struct {
-	Key   string
-	Value string
+type OptionalData struct {
```

Comment:
I think it would be better to rename this an `OptionalDataEntry` instead, since it represents a single entry inside the optional data array. 

Then, we can define the type `OptionalData` as 

```
type OptionalData []OptionalDataEntry
```